### PR TITLE
Insert transposes around view_copy ops

### PR DESCRIPTION
**Pull request by Erik-Lundell**
Change-Id: Ic15088557542ff6c3c97bda2cb0bd3ab642e00ef

Change-Id: I0c64f6f92d4173420af0a6cf2d707c63beeeaa00

### CODE REVIEW

File: backends/arm/test/ops/test_layer_norm.py [hunk 1]
```diff
@@ -148,7 +148,8 @@ def test_layer_norm_tosa_BI(
             self.LayerNorm(*model_params), (test_data,)
         )
 
-    @parameterized.expand(test_data_suite)
+    # Skip last test since it requires transpose.
```
Comment (digantdesai) [inline backends/arm/test/ops/test_layer_norm.py:151]: and U55 doesn't support it.

**Comment by digantdesai** [comment]
curious if there is a copy cost of doing these? is there a value in optimizing these nodes we are inserting here

**Review by digantdesai — APPROVED** [review]

**Comment by Erik-Lundell** [comment]
> curious if there is a copy cost of doing these? is there a value in optimizing these nodes we are inserting here

There is a copy cost, and we also want to minimize transpose use since it's not always supported. I came up with a criteria that is a bit smarter than always inserting them in this PR but it might be possible to reduce them even more. We have discussed introducing some way of lazingly transposing just when it's needed but we are focusing on getting things functional first.

**Comment by digantdesai** [comment]
> > curious if there is a copy cost of doing these? is there a value in optimizing these nodes we are inserting here
> 
> There is a copy cost, and we also want to minimize transpose use since it's not always supported. I came up with a criteria that is a bit smarter than always inserting them in this PR but it might be possible to reduce them even more. We have discussed introducing some way of lazingly transposing just when it's needed but we are focusing on getting things functional first.

Sounds good. We do something greedy on XNNPACK side to reduce number of format switches.

**Pull request merged**